### ISSUE TITLE
AI tie-in to app

### ISSUE THREAD

**Issue by davideliason** [Thitiat-Pidtatasang/ToryBurch_Github#5]
One approach: S3 bucket > document chunks > embeddigs model > vector dB (MongodB or Pinecone (free)
**RAG**= allows a Foundation Model to ref data source outside its training data (S3, Salesforce, web pages, etc). Basically the external data from Pinecone or DB is used to enhance the generic Foundation Model response. So we add company-specific info in the DB/Pinecone
- not as cheap as Prompt Engineering but the next best thing, cheaper than Fine-Tuning
- the s3 bucket data is added to Knowledge Base from Vector Database and added to FM's prompt
- create chatbot with specific knowledge that we add about Tory Burch and products
Bedrock has **Guardrails** service that removes PII (ie customer database info)
Bedrock **Agent**: carry out multi-step tasks, leverage RAG, integrate with DB and API to take action or get more info
- the Agent works with API Lambda, Knowledge Base to access DB data
**CloudWatch**: logging and metric integration with Bedrock
Save $ with Bedrock: small model, batch (50% off), lower # of I/O tokens
